sdk/python/lib/pulumi/resource.py

```diff
@@ -482,6 +482,11 @@ class ResourceOptions:
     require replacement instead of update only if `"*"` is passed.
     """
 
+    retain_on_delete: Optional[bool]
+    """
+    Option to not actually delete resources from providers when the engine calls Delete.
```

Comment:
Nit: try to explaining this in terms of what the user gets not how it's implemented. Taking a shot at it:

"If set to True, deletions retain the resource it the cloud instead of deleting it. Pulumi simply stops tracking the deleted resource."